packages/mobile/src/screens/sign-on-screen/components/SignupFlowTikTokAuth.tsx

```diff
@@ -0,0 +1,94 @@
+import { useEffect } from 'react'
+
+import {
+  useAudiusQueryContext,
+  type TikTokProfileData,
+  pickHandleSchema,
+  Name
+} from '@audius/common'
+import { useDispatch, useSelector } from 'react-redux'
+import { restrictedHandles } from 'utils/restrictedHandles'
+
+import { TikTokAuthButton } from 'app/components/tiktok-auth'
+import { make, track } from 'app/services/analytics'
+import * as oauthActions from 'app/store/oauth/actions'
+import { getAbandoned } from 'app/store/oauth/selectors'
+
+type SignupFlowTikTokAuthProps = {
+  onStart: () => void
+  onFailure: (e: unknown) => void
+  onSuccess: (info: {
+    requiresReview: boolean
+    handle: string
+    platform: 'tiktok'
+  }) => void
+  onClose: () => void
+}
+
+// Wrapper around TikTokAuthButton that
+export const SignUpFlowTikTokAuth = ({
```

Comment:
I don't love this abstraction around the `TikTokAuthButton` but for the sake of backwards compatibility I didn't want to mess with the existing component and instead just add a wrapper (this is also how we have it on web rn)